tools/infer/utility.py

```diff
@@ -245,8 +249,9 @@ def create_predictor(args, mode, logger):
                     logger.info(E)
                     logger.info("Please keep your paddlepaddle-gpu >= 2.3.0!")
 
-        elif args.use_npu:
-            config.enable_custom_device("npu")
+        elif args.use_custom_device:
+            dev_type = paddle.device.get_all_custom_device_type()[0]
+            config.enable_custom_device(dev_type)
```

Comment:
The custom device should not be able to replace npu.

It seems you should add mlu device on top of the original code with another elif clause.

Reply:
config.enable_custom_device("mlu")